consensus/src/dag/types.rs

```diff
@@ -288,6 +284,16 @@ impl Deref for CertifiedNode {
     }
 }
 
+impl TDAGMessage for CertifiedNode {
+    fn verify(&self, verifier: &ValidatorVerifier) -> anyhow::Result<()> {
+        let node_digest = NodeDigest::new(self.digest());
```

Comment:
we can't use the digest value, we need to calculate the digest for verification

Reply:
In #8971